### ISSUE TITLE
[select] use arrow keys to open dropdown instead of change value

### ISSUE THREAD

**Issue by DRiFTy17** [tyler-technologies-oss/forge#244]
**Describe the bug:**
The arrow keys should be used to open the dropdown instead of changing the value of the field immediately.

One issue with the current implementation is that screen readers do not announce the change in this scenario either. So if we do decide to keep this "feature" then we need to ensure that value is properly announced.

Another thing to note is that we allow for typing characters to naively "filter" the options while the dropdown is closed. This does not announce either and it should.

**To Reproduce:**
1. Tab to <forge-select>
2. Use up or down arrow keys to change value
3. Observe that screen reader does not announce the value changing

**Expected behavior:**

1. The <forge-select> should open the dropdown with the up or down arrow keys instead of immediately changing the value.
2. It should also announce value changes when the dropdown is closed
